main.py

```diff
@@ -31,11 +38,194 @@ def setup(self):
+        # Keys necessary to delete a flow, 'cookie_mask' is added later
+        keys = {'idle_timeout', 'hard_timeout', 'cookie', 'priority',
+                'match', 'actions'}
+        command = "v2/stored_flows?state=installed"
+        response = requests.get(f"{FLOW_MANAGER_URL}/{command}")
+
+        if response.status_code // 100 != 2:
+            log.error(f"Could not get the flows from flow_mager. Status "
+                      f"code {response.status_code}.")
+            # Problem here: pipeline stays as 'enabling', error needs
+            # to be handled. Events have been sent and NApps are probably
+            # working in a different pipeline setup
+            return
+        if pipeline.get("status") is None:
+            # Changing to default pipeline. Miss flow entries are not needed
+            self.delete_miss_flows()
+        flows_by_swich = response.json()
+        set_up = self.build_content(pipeline)
+
+        for switch in flows_by_swich:
+            delete_flows = {"flows": []}
+            install_flows = {"flows": []}
+            for flow in flows_by_swich[switch]:
+                owner = flow["flow"]["owner"]
+                if owner not in set_up:
+                    continue
+                expected_table_id = set_up[owner][flow["flow"]["table_group"]]
+                # if table_id needs to change
+                if expected_table_id != flow["flow"]["table_id"]:
+                    # Get key-value from flow to be sent to flow_manager
+                    delete = {key: flow["flow"].get(key) for key in keys}
+                    delete["cookie_mask"] = flow["flow"]["cookie"]
+                    delete_flows["flows"].append(delete.copy())
+
+                    # Change table_id before being added
+                    flow["flow"].update({"table_id": expected_table_id})
+                    install_flows["flows"].append(flow["flow"].copy())
+
+            if delete_flows["flows"]:
+                self.send_flows(delete_flows, switch, "delete")
+            if install_flows["flows"]:
+                self.send_flows(install_flows, switch, "add")
+            self.install_miss_flows(pipeline, switch)
+        if pipeline.get('id'):
+            self.pipeline_controller.update_status(pipeline["id"], "enabled")
+
+    def install_miss_flows(self, pipeline: dict, switch: str):
+        """Install miss flow entry to a switch"""
+        install_flows = {"flows": []}
+        cookie = self.get_cookie(switch)
+        for table in pipeline["multi_table"][:-1]:
+            instruction = [{"instruction_type": "goto_table",
+                            "table_id": table['table_id'] + 1}]
+            flow = {
+                'priority': table.get('priority', 0),
+                'cookie': cookie,
+                'instructions': table.get('instructions', instruction),
+                'owner': 'of_multi_table',
+                'table_group': 'base',
+                'table_id': table['table_id'],
```

Comment:
I am installing EPL as miss flow entries. Planning to install a miss flow only when the pipeline has it.

Reply:
Right, yes it should only be installing when `table_miss_flow` isn't an empty. I'll review this part again in the feature when you update it.

Reply:
Now only installing miss flows if any is defined in the pipeline.